cmd/antrea-agent/agent.go

```diff
@@ -418,13 +420,21 @@ func run(o *Options) error {
 
 	// Start the Antrea IPAM agent.
 	if features.DefaultFeatureGate.Enabled(features.AntreaIPAM) {
+		listOptions := func(options *metav1.ListOptions) {
+			options.FieldSelector = fields.OneTermEqualSelector("spec.nodeName", nodeConfig.Name).String()
+		}
+		localNodeInformerFactory := informers.NewSharedInformerFactoryWithOptions(k8sClient, informerDefaultResync, informers.WithTweakListOptions(listOptions))
```

Comment:
Why we need a new InformerFactory here, rather than reusing informerFactory?

Reply:
We need `listOptions` here to cache local Pods only.

Reply:
For agent, I think we should not cache all the Pods.

Reply:
Could you check InitializeNPLAgent()? It has the same requirement but reuses informerFactory.

Reply:
I already checked InitializeNPLAgent and the previous commit uses this method (`NewFilteredPodInformer`, but the resyncPeriod is different).
And now I moved to `NewSharedInformerFactoryWithOptions`.
```
	listOptions := func(options *metav1.ListOptions) {
		options.FieldSelector = fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
	}
	podInformer := coreinformers.NewFilteredPodInformer(
		kubeClient,
		metav1.NamespaceAll,
		resyncPeriod,
		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, // NamespaceIndex is used in NPLController.
		listOptions,
	)
```

Reply:
Could you explain more: "but the resyncPeriod is different"?

Reply:
The resyncPeriod is set to "0" in npl_agent_init.go for podInformer, but the resyncPeriod is 12 hours for informerFactory by default.
Thus I didn't make IPAM controller and NPL controller to share one podInformer to cache local Pods.

Reply:
I thought we can create a new PodInformer using the same InformerFactory. Probably I am wrong?
@tnqn should chime in here.

Reply:
It doesn't need to create another informerFactory for a single pod informer. Creating an informerFactory only makes sense when you need to create more than multiple informers, especially multiple instances per kind, to make them share the cache.
We can just create the localPodInformer on demand directly (without creating the factory first) here and pass it to NPLController and IPAMController.
I think NPLController and IPAMController don't need any resync so `resyncPeriod` can be 0 for both cases.

Reply:
From a single InformerFactory, we cannot create two informers for the same resource type?

Reply:
It can. I meant as the filter condition of Pod is different from other resource types, we cannot reuse the existing informerFactory. But there is no need to create another informer factory just for podInformer (no other resource types will reuse this informer factory as filtering by node only makes sense for Pod), it could just create the informer directly and pass it to the two consumers.

Reply:
Filter (ListOptions) can be configured when creating the Informer with NewFilteredPodInformer right. For example:

```
podInformer := coreinformers.NewFilteredPodInformer(
		kubeClient,
		metav1.NamespaceAll,
		resyncPeriod,
		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc}, // NamespaceIndex is used in NPLController.
		listOptions,
	)
```

Reply:
Yes, that's how we create the "localPodInfomer" I meant. Note that it is not created from any informer factory but from kubeClient directly. My suggestion was moving this code to cnd/antrea-agent/agent.go, no NPLController and IPAMController can share it.

Reply:
Changed to one localPodInformer for AntreaIPAM and NPL agent.